### DISCUSSION TITLE
Support `--https` for `gatsby serve`

### DISCUSSION THREAD

**Discussion by techfg** [gatsbyjs/gatsby#38529] [Ideas / Feature Requests]
# Background

`gatsby develop` supports the `--https` option, whereas `gatsby serve` does not.  Often times, being able to test a "production" build with HTTPS is critical (e.g., testing security headers).

# Approach

Add parameters to `serve` command and modify serve.js to follow similar `getSslCert` approach that `develop` process does, creating an httpsServer to listen.

Thoughts/input welcome!

**Comment by jsloyer** [comment]
I would love this myself.  I would be happy to contribute the code for this if others would be receptive to this going in.

This would be nice to have with the Gatsby cli as the express integration is pretty out of date.  With this enhancement it will make it easier to run Gatsby inside of Kubernetes.

**Comment by jsloyer** [comment]
Opened up a RFC at https://github.com/gatsbyjs/gatsby/discussions/38690
